desktop/core/src/desktop/models.py

```diff
@@ -959,7 +959,10 @@ def documents(self, user, perms='both', include_history=False, include_trashed=F
     if not include_trashed:
       docs = docs.exclude(is_trashed=True)
 
-    return docs.defer('description', 'data', 'extra', 'search').distinct().order_by('-last_modified')
+    if allow_distinct:
+      return docs.defer('description', 'data', 'extra', 'search').order_by('-last_modified')
+    else:
+      return docs.defer('description', 'data', 'extra', 'search').distinct().order_by('-last_modified')
```

Comment:
not sure on this @romainr

Reply:
Yes, just avoid the duplication:

e.g.

```
if allow_distinct:
  docs = docs.distinct()
```

and obviously you reversed the conditions above